src/hydra/HydraAdmin.js

```diff
@@ -1,50 +1,165 @@
-import React from 'react';
+import React, {useEffect, useState} from 'react';
 import PropTypes from 'prop-types';
-import {Provider} from 'react-redux';
+import {AdminContext, AdminUI, Error as ErrorUI, Loading} from 'react-admin';
 import {createHashHistory} from 'history';
-import withContext from 'recompose/withContext';
-import {createAdminStore} from 'react-admin';
+import {createMuiTheme} from '@material-ui/core/styles';
 
 import dataProviderFactory from './dataProvider';
-import AdminGuesser from '../AdminGuesser';
+import ResourceGuesser from '../ResourceGuesser';
+import {Layout} from '../layout';
 
-const history = createHashHistory();
+const displayOverrideCode = resources => {
+  if (process.env.NODE_ENV === 'production') return;
+
+  let code =
+    'If you want to override at least one resource, paste this content in the <HydraAdmin> component of your app:\n\n';
+
+  resources.forEach(r => {
+    code += `<ResourceGuesser name={"${r.name}"} />\n`;
+  });
+  console.info(code);
+};
+
+/**
+ * AdminGuesser automatically renders an `<AdminUI>` component for resources exposed by a web API documented with Hydra, OpenAPI or any other format supported by `@api-platform/api-doc-parser`.
+ * If child components are passed (usually `<ResourceGuesser>` or `<Resource>` components, but it can be any other React component), they are rendered in the given order.
+ * If no children are passed, a `<ResourceGuesser>` component is created for each resource type exposed by the API, in the order they are specified in the API documentation.
+ */
+export const AdminGuesser = ({
+  children,
+  includeDeprecated,
+
+  resources,
+  fetching,
+  error,
+
+  ...rest
+}) => {
+  if (fetching) {
+    return <Loading />;
+  }
+
+  if (error) {
+    if ('production' === process.env.NODE_ENV) {
+      console.error(error);
+    }
+
+    return (
+      <ErrorUI
+        error={new Error('API schema is not readable')}
+        errorInfo={{componentStack: null}}
+      />
+    );
+  }
+
+  if (!children && resources) {
+    const guessResources = includeDeprecated
+      ? resources
+      : resources.filter(r => !r.deprecated);
+    children = guessResources.map(r => (
+      <ResourceGuesser name={r.name} key={r.name} />
+    ));
+    displayOverrideCode(guessResources);
+  }
+
+  return <AdminUI {...rest}>{children}</AdminUI>;
+};
 
 const HydraAdmin = ({
   entrypoint,
   dataProvider = dataProviderFactory(entrypoint),
-  i18nProvider,
   authProvider,
+  i18nProvider,
+  history = createHashHistory(),
```

Comment:
WDYT about making it configurable to support also browser history?

Reply:
It's a prop so you can pass whatever you want (for example an instance of `createBrowserHistory`).

Reply:
You should add info about this change to "Things done:" because it was not possible before :)

Reply:
Sure :slightly_smiling_face: